frontend/src/lib/components/CommandBar/CommandBar.tsx

```diff
@@ -0,0 +1,54 @@
+import { useRef, forwardRef } from 'react'
+import { useActions, useValues } from 'kea'
+
+import { useEventListener } from 'lib/hooks/useEventListener'
+import { useOutsideClickHandler } from 'lib/hooks/useOutsideClickHandler'
+
+import { commandBarLogic } from './commandBarLogic'
+import { BarStatus } from './types'
+
+import './index.scss'
+import SearchBar from './SearchBar'
+import { LemonModal } from '@posthog/lemon-ui'
+
+const CommandBarContainer = forwardRef<HTMLDivElement, { children?: React.ReactNode }>(function CommandBarContainer(
+    { children },
+    ref
+): JSX.Element {
+    return (
+        <div className="w-full h-160 max-w-lg bg-bg-3000 rounded overflow-hidden flex flex-col" ref={ref}>
+            {children}
+        </div>
+    )
+})
+
+function CommandBar(): JSX.Element | null {
+    const containerRef = useRef<HTMLDivElement | null>(null)
+    const { barStatus } = useValues(commandBarLogic)
+    const { toggleSearchBar, toggleActionsBar, hideCommandBar } = useActions(commandBarLogic)
+
+    useEventListener('keydown', (event) => {
+        if ((event.ctrlKey || event.metaKey) && event.key === 'k') {
+            event.preventDefault()
+            if (event.shiftKey) {
+                toggleActionsBar()
+            } else {
+                toggleSearchBar()
+            }
+        } else if (event.key === 'Escape') {
+            hideCommandBar()
+        }
+    })
+
+    useOutsideClickHandler(containerRef, hideCommandBar, [])
+
+    return (
+        <LemonModal isOpen={barStatus !== BarStatus.HIDDEN} simple closable={false} width={800}>
```

Comment:
Closely related to the slugishness point, I'd say the command palette explicitly should _not_ be a modal. It should be a lightweight `fixed` overlay. E.g. when you use Spotlight, Raycast, Cmd+Shift+P in VS Code, or Cmd+T in Arc Browser, the palette shows up instantly and doesn't block interactions with whatever's below. That feels snappier than an animated modal which takes over 100% of the experience.